discovery.md

```diff
@@ -521,14 +545,190 @@ When encoded in JSON, the response format MUST adhere to the following:
+The `id` MUST be generated by the Discovery Enpoint.
+
+An error MUST be generated if the request can not be fulfilled.
+
+The follow responses are defined by this specification:
+- `201 Created` if the new Service was added. The response MUST include
+  an HTTP `Location` Header with a value that points to the new Service.
+  The HTTP Response Body MUST include the JSON represenation of the
+  newly created Service.
+- `202 Accepted` to indicate that the request has been accepted but not
+  processed yet. The response MUST include an HTTP `Location` Header with
+  a value that can be used to determine the status of the request. The
+  HTTP Response Body MAY be empty.
+- `406 Not Acceptable` to indicate that the request message is syntactically
+  invalid.  For example, if it contains an `id` attribute.
+  The HTTP Response Body SHOULD contain information as to why the create
+  failed.
+- `409 Conflict` to indicate that the request message is syntactically valid,
+  but the value of one or more of the attributes are invalid - for example, the
+  `name` is not unique.
+  The HTTP Response Body SHOULD contain information as to why the create
+  failed.
+
+Other responses are allowed, but not defined by this specification.
+
+If a `202 Accepted` was returned, an HTTP `GET` to the `Location` endpoint
+can be used to determine the status of the original request.
+This specification defines the following responses:
+- `200 OK` and the new Service representation in the HTTP Response Body -
+  meaning that this URL is the final URL of the newly added Service. Note:
+  in this case the URL MUST adhere to the normal `/services/{id}` format.
+- `204 No Content` to indicate that the `POST` request is not completed yet.
+  The HTTP Response Body MAY be empty.
+- `301 Moved Permanently` to indicate that the Service was successfully
+  added and can be found at the URL in the REQUIRED `Location` header of the
+  response.
+- `406 Not Acceptable` to indicate that the original request message was
+  syntactically invalid.  For example, if it contains an `id` attribute.
+  The HTTP Response Body SHOULD contain information as to why the create
+  failed.
+- `409 Conflict` to indicate that the original request message is
+  syntactically valid, but the value of one or more of the attributes are
+  invalid - for example, the `name` is not unique.
+  The HTTP Response Body SHOULD contain information as to why the create
+  failed.
+
+Other responses are allowed, but not defined by this specification, however
+they are related to the `GET` itself and not the original add request.
+
+How long the Discovery Endpoint will support the original `Location` URL is
+an implementation choice, however, it MUST be available immediately upon
+return of the `202 Accepted`. Meaning, a `404 Not Found` MUST NOT be returned
+in response to a `GET` to the `Location` URL during the time the creation
+request is being processed.
+
+##### `PUT /services/{id}`
```

Comment:
I would break this into another interface, creating services is not about discovering them.

Reply:
yes I'm working on putting the management stuff into a separate section - just couldn't get to it last night...